form.yml

```diff
@@ -0,0 +1,329 @@
+
+  sapmnt_inst_media:
+    $name: NFS share containing Netweaver SID folder
+    $type: text
+    $help: NFS share with the SID folder
+    $optional: false
+
+  use_swpm_sar_file:
+    $name: Provide SWPM installer sar archive for Netweaver installation 
+    $type: boolean
+    $default: false
+    $help: Mark this option if you want to use swpm sar archive file for the installation instead of already extracted SWPM installer
+  swpm_sar_file:
+    $name: Path to SWPM archive
+    $visibleIf: .use_swpm_sar_file == true
+    $type: text
+    $help: The path to swpm installer media archive in SAR format. You also need to provide the sapcar executable path for extraction
+    $optional: true
+  nw_extract_dir:
+    $name: Path to extract the Netweaver installation media
+    $visibleIf: .use_swpm_sar_file == true
+    $type: text
+    $default: /sapmedia_extract/NW
+    $help: The NW media will be extracted to this path. By default this path is /sapmedia_extract/NW
+    $optional: true
+  sapcar_exe_file:
+    $name: Path to sapcar executable if extracting NW SAR archive
+    $visibleIf: .use_swpm_sar_file == true
+    $type: text
+    $help: The path to sapcar executable to extract NW SAR archive
+    $optional: true
+
+  swpm_folder:
+    $name: SWPM folder location
+    $visibleIf: .use_swpm_sar_file == false
+    $type: text
+    $help: Path to already extracted SWPM installer folder
+    $optional: true
+  
+  sapexe_folder:
+    $name: SAP exe folder location
+    $type: text
+    $help: Path to sap exe folder. This folder must contain all the downloaded software from SAP download center to install the Netweaver systems. like SAPEXE, SAPEXEDB, igshelper, etc
+    $optional: false
+  
+  installation_folder:
+    $name: Netweaver installation folder
+    $type: text
+    $default: /tmp/swpm_unattended
+    $help: Folder where the installation is executed. It must be an empty folder. The execution logs will be stored there
+    $optional: false
+  
+  ha_enabled:
+    $name: Enable HA cluster configuration
+    $type: boolean
+    $default: true
+    $help: Enable the HA cluster configuration. In order to make this work, ASCS and ERS system must be installed together in parallel.
+    $optional: false
+
+  nfs_version:
+    $name: NFS version
+    $type: text
+    $default: nfs4
+    $help: Used to connect to the nfs share
+    $optional: true
+
+  nfs_options:
+    $name: NFS options
+    $type: text
+    $default: defaults
+    $help: Specify the options for nfs share
+    $optional: true
+
+  additional_dvds:
+    $name: Additional dvds
+    $type: edit-group
+    $help: Path to software folder or archives with additional SAP software
+    $minItems: 1
+    $prototype:
+        $name: Additional DVD
+        $type: text
+        $optional: false
+
+  saptune_solution:
+    $name: saptune solution to apply
+    $type: text
+    $default: NETWEAVER
+    $help: saptune solution to apply to all nodes
+    $optional: true
+
+  hana_checkbox:
```

Comment:
We say that is mandatory only for `DB/PAS/AAS` but later all of the are options, right? We should change this

Reply:
@arbulu89 I did not understand. Do you mean that this entry should not appear for `ERS/ASCS` instances? In that case, we can detect what type of instance user enters, unless we provide a drop-down option to user to choose the instance type. With this drop-down we can also show/hide some section based on instance type selection.

or you mean we need to set the `$optional: true` for hana sub entries

Reply:
Yes, I got it. In the form, I think we can implement the logic via drop-down